x-pack/plugins/ml/public/application/components/job_spaces_selector/spaces_selectors.tsx

```diff
@@ -0,0 +1,220 @@
+import { useSpacesContext } from '../../contexts/spaces';
+import { ML_SAVED_OBJECT_TYPE } from '../../../../common/types/saved_objects';
+import { ALL_SPACES_ID } from '../job_spaces_list';
+import { CannotEditCallout } from './cannot_edit_callout';
+
+type SpaceOption = EuiSelectableOption & { ['data-space-id']: string };
+
+interface Props {
+  jobId: string;
+  spaceIds: string[];
+  setSelectedSpaceIds: (ids: string[]) => void;
+  selectedSpaceIds: string[];
+  canEditSpaces: boolean;
+  setCanEditSpaces: (canEditSpaces: boolean) => void;
+}
+
+export const SpacesSelector: FC<Props> = ({
+  jobId,
+  spaceIds,
+  setSelectedSpaceIds,
+  selectedSpaceIds,
+  canEditSpaces,
+  setCanEditSpaces,
+}) => {
+  const { spacesManager, allSpaces } = useSpacesContext();
+
+  const [canShareToAllSpaces, setCanShareToAllSpaces] = useState(false);
+
+  useEffect(() => {
+    const getPermissions = spacesManager.getShareSavedObjectPermissions(ML_SAVED_OBJECT_TYPE);
+    Promise.all([getPermissions]).then(([{ shareToAllSpaces }]) => {
+      setCanShareToAllSpaces(shareToAllSpaces);
+      setCanEditSpaces(shareToAllSpaces || spaceIds.includes(ALL_SPACES_ID) === false);
+    });
+  }, []);
+
+  function toggleShareOption(isAllSpaces: boolean) {
+    const updatedSpaceIds = isAllSpaces
+      ? [ALL_SPACES_ID, ...selectedSpaceIds]
+      : selectedSpaceIds.filter((id) => id !== ALL_SPACES_ID);
+    setSelectedSpaceIds(updatedSpaceIds);
+  }
+
+  function updateSelectedSpaces(selectedOptions: SpaceOption[]) {
+    const ids = selectedOptions.filter((opt) => opt.checked).map((opt) => opt['data-space-id']);
+    setSelectedSpaceIds(ids);
+  }
+
+  const isGlobalControlChecked = useMemo(() => selectedSpaceIds.includes(ALL_SPACES_ID), [
+    selectedSpaceIds,
+  ]);
+
+  const options = useMemo(
+    () =>
+      allSpaces.map<SpaceOption>((space) => {
+        return {
+          label: space.name,
+          prepend: <SpaceAvatar space={space} size={'s'} />,
+          checked: selectedSpaceIds.includes(space.id) ? 'on' : undefined,
+          disabled: canEditSpaces === false,
+          ['data-space-id']: space.id,
+          ['data-test-subj']: `mlSpaceSelectorRow_${space.id}`,
+        };
+      }),
+    [allSpaces, selectedSpaceIds, canEditSpaces]
+  );
+
+  const shareToAllSpaces = useMemo(
+    () => ({
+      id: 'shareToAllSpaces',
+      title: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.title', {
+        defaultMessage: 'All spaces',
+      }),
+      text: i18n.translate('xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.text', {
+        defaultMessage: 'Make job available in all current and future spaces.',
+      }),
+      ...(!canShareToAllSpaces && {
+        tooltip: isGlobalControlChecked
+          ? i18n.translate(
+              'xpack.ml.management.spacesSelectorFlyout.shareToAllSpaces.cannotUncheckTooltip',
+              { defaultMessage: 'You need additional privileges to change this option.' }
```

Comment:
>  { defaultMessage: 'You need additional privileges to change this option.' }

Will we be able to provide more info about which privileges are required (or where they can find that info)?

Reply:
We could possibly have a link to online documentation. Privileges are so complicated that I don't think we could provide a useful summary here.
Also, this text is a verbatim copy of the message on the saved objects manager. 
https://github.com/elastic/kibana/blob/8b658fbcd2e97546b59a156df57b02d866882710/x-pack/plugins/spaces/public/share_saved_objects_to_space/components/share_mode_control.tsx#L86

Reply:
> Privileges are so complicated that I don't think we could provide a useful summary here.

Agreed. During review of the original code in the Share to Space flyout, we decided to keep this statement brief, since it's a tooltip.

The callout mentioned in https://github.com/elastic/kibana/pull/83320#discussion_r524810873 above is also displayed at the same time as this tooltip icon, IMO that provides an opportunity to be more descriptive.

Reply:
Agreed with keeping this simple.  To link to the docs, you can use:

You need additional privileges to change this option.  [Learn more](url).